### PR TITLE
AUT-1705: Adding SonarQube for react-client.

### PR DESCRIPTION
**Pull request by gpochettino**
# React SDK

## What did you accomplish?
Adding SonarQube step to CI in order have a check for code quality

Since main and development are two permanent branches, we are having two executions in order to make sonar to know about the parent branch.

## How do we test the changes introduced in this PR?
Testing from this PR if the step is being executed and giving the right output.

From SonarQube site this PR is being analyzed and showing code coverage.
https://sonarqube.split-internal.com/dashboard?id=react-client&pullRequest=118
Screen Shot 2022-12-15 at 10 02 46

Enabled decoration for PR:
Screen Shot 2022-12-15 at 10 02 58

## Extra Notes

Node version using lts/* was bringing node 18, so that caused problems to build. I updated it to 16.16.0 which was working. I did that since I checked from previous successful builds where lts/* was pulling node 16.16.0. 

In order to get the blame information to SonarQube, we told GitHubActions to clone the entire history. This was done on this section: 
```
      - name: Checkout code
        uses: actions/checkout@v3
        with:
          fetch-depth: 0
```

From SonarQube site I applied the following changes:

Set master & development branches as active branches.
Enabled PullRequest decoration

### CODE REVIEW

File: .github/workflows/ci-cd.yml [hunk 1]
```diff
@@ -18,22 +18,24 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - name: Checkout code
-        uses: actions/checkout@v2
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
 
       - name: Set up nodejs
         uses: actions/setup-node@v2
```
Comment (melinaurruchua) [inline .github/workflows/ci-cd.yml:26]: actions/setup-node should be updated to v3

File: .github/workflows/ci-cd.yml [hunk 2]
```diff
@@ -18,22 +18,24 @@ jobs:
...
         with:
-          node-version: 'lts/*'
+          node-version: '16.16.0'
           cache: 'npm'
-
+
```
Comment (melinaurruchua) [inline .github/workflows/ci-cd.yml:30]: can we remove the extra spaces?

File: .github/workflows/ci-cd.yml [hunk 3]
```diff
@@ -69,3 +71,37 @@ jobs:
           SOURCE_DIR: ./umd
           DEST_DIR: sdk
           ARGS: --acl public-read --follow-symlinks --cache-control max-age=31536000,public
+
+
+      - name: SonarQube Scan (Push)
+        if: github.event_name == 'push'
+        uses: SonarSource/sonarcloud-github-action@v1.6
```
Comment (melinaurruchua) [inline .github/workflows/ci-cd.yml:78]: SonarSource/sonarcloud-github-action is on 1.8, we can consider an update too

**Review by melinaurruchua — CHANGES_REQUESTED** [review]

File: .github/workflows/ci-cd.yml (hunk 1 above)
Comment (gpochettino) [reply .github/workflows/ci-cd.yml]: Thanks Meli for pointing this out!

File: .github/workflows/ci-cd.yml (hunk 3 above)
Comment (gpochettino) [reply .github/workflows/ci-cd.yml]: Thanks! I updated it!

File: .github/workflows/ci-cd.yml (hunk 2 above)
Comment (gpochettino) [reply .github/workflows/ci-cd.yml]: Remove them :D

**Review by melinaurruchua — DISMISSED** [review]

**Review by EmilianoSanchez — APPROVED** [review]

**Review by EmilianoSanchez — APPROVED** [review]

**Pull request merged**